dask/base.py

```diff
@@ -139,11 +141,20 @@ def normalize_function(func):
 
 
 normalize_token = Dispatch()
-normalize_token.register((int, float, str, tuple, list), lambda a: a)
-normalize_token.register(object,
-        lambda a: normalize_function(a) if callable(a) else a)
+normalize_token.register((int, float, str, unicode, bytes, tuple, list,
```

Comment:
Sorry for chiming in so late here. I'm -0.5 on descending into containers, as I expect large performance penalities in code using `dask.imperative`. For example, suppose a user passes a large list of tuples to a `do` wrapped function. This will now have N*M dispatches just for that list, rather than a single `str` call - resulting in much slower behavior.